tabular/src/autogluon/tabular/models/tab_transformer/hyperparameters/parameters.py

```diff
@@ -0,0 +1,64 @@
+                    'encoders':  {'CATEGORICAL': 'CategoricalOrdinalEnc', # How to "encode"(vectorize) each column type.
+                                  'DATETIME'   : 'DatetimeOrdinalEnc',
+                                  'LATLONG'    : 'LatLongQuantileOrdinalEnc',
+                                  'SCALAR'     : 'ScalarQuantileOrdinalEnc',
+                                  'TEXT'       : 'TextSummaryScalarEnc'},
+                    'aug_mask_prob' : 0.4, # What percentage of values to apply augmentation to.
+                    'num_augs' : 1, # Number of augmentations to add.
+                    'pretext': 'BERT_pretext', # What pretext to use when performing pretraining/semi-supervised learning.
+                    'n_cont_features': 8, # How many continuous features to concatenate onto the categorical features
+                    'fix_attention': False, # If True, use the categorical embeddings in the transformer architecture.
+                    'epochs': 100, # How many epochs to train on with labeled data.
+                    'pretrain_epochs': 200, # How many epochs to pretrain on with unlabeled data.
+                    'epochs_wo_improve': 50, # How many epochs to continue running without improving on metric. aka "Early Stopping Patience"
+                    'num_workers': 16, # How many workers to use for torch DataLoader.
+                    'max_columns': 1000, # Maximum number of columns TabTransformer will accept as input. This is to combat huge memory requirements/errors.
+                    'num_output_layers': 1 # How many fully-connected layers on top of transformer to produce predictions. Minimum 1 layer.
+                    }
+
+    return fixed_params
+
+def get_hyper_params():
+    """ Parameters that currently can be tuned during HPO """
+    hyper_params = {
+        'lr': 1e-3, # Learning rate
+        # Options: Real(5e-5, 5e-3)
+        'weight_decay': 1e-6, # Rate of linear weight decay for learning rate
+        # Options: Real(1e-6, 5e-2)
+        'p_dropout': 0.1, # dropout probability, 0 turns off Dropout.
+        # Options: Categorical(0, 0.1, 0.2, 0.3, 0.4, 0.5)
+        'n_heads': 8, # Number of attention heads
+        # Options: Categorical(2, 4, 8)
+        'hidden_dim': 128, # hidden dimension size
+        # Options: Categorical(32, 64, 128, 256)
+        'n_layers': 1, # Number of Tab Transformer encoder layers,
+        # Options: Categorical(1, 2, 3, 4, 5)
+        'feature_dim': 64, # Size of fully connected layer in TabNet.
+        # Options: Int(8, 128)
+        'tab_readout': 'none', # What sort of readout from the transformer.
+        # Options: ['readout_emb', 'mean', 'concat_pool', 'concat_pool_all', 'concat_pool_add', 'all_feat_embs', 'mean_feat_embs', 'none']
+    }
+
+    return hyper_params
+
+def get_default_param(problem_type):
+
+    params = get_fixed_params()
+    params['problem_type'] = problem_type
```

Comment:
Probably remove

Reply:
Line 61?
I need 'problem_type' in other places like pretexts/utils. Is it worth taking it out here, but adding additional parameters in TabTransformerDataset/SUPERVISED_pretext?
Overall design question as well:
Could potentially remove the whole kwargs passing to both of these, but would need to add 2-3 params when creating. Which is preferred?

Reply:
Removed.